### PR TITLE
Refactor Gen 2 Quickstart to have Pages and App Router implementations

### PR DESCRIPTION
**Pull request by kevinold**
#### Description of changes:

Refactor Gen 2 Quickstart to have Pages and App Router implementations

#### Related GitHub issue #, if available:

https://app.asana.com/0/1205212189033571/1205977177296272/f

### Instructions

**If this PR should not be merged upon approval for any reason, please submit as a DRAFT**

Which product(s) are affected by this PR (if applicable)?
- [ ] amplify-cli
- [ ] amplify-ui
- [ ] amplify-studio
- [ ] amplify-hosting
- [ ] amplify-libraries

Which platform(s) are affected by this PR (if applicable)?
- [x] JS
- [ ] iOS
- [ ] Android
- [ ] Flutter
- [ ] React Native

**Please add the product(s)/platform(s) affected to the PR title**

#### Checks

- [ ] Does this PR conform to [the styleguide](https://github.com/aws-amplify/docs/blob/main/STYLEGUIDE.md)?

- [ ] Does this PR include filetypes other than markdown or images? Please add or update unit tests accordingly.

- [ ] Are any files being deleted with this PR? If so, have the needed redirects been created?

- [ ] Are all links in MDX files using the MDX link syntax rather than HTML link syntax? <br /> 
      _ref: MDX: `[link](https://link.com)` 
            HTML: `<a href="https://link.com">link</a>`_

### When this PR is ready to merge, please check the box below
- [x] Ready to merge

_By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license._

### CODE REVIEW

File: src/pages/gen2/start/quickstart/nextjs-app-router/index.mdx [hunk 1]
```diff
@@ -0,0 +1,519 @@
...
+
+```ts title=components/ConfigureAmplify.tsx
+// components/ConfigureAmplify.tsx
+
+"use client";
+
+import config from "@/amplifyconfiguration.json";
+import { Amplify } from "aws-amplify";
+
+Amplify.configure(config as any, { ssr: true });
```
Comment (josefaidt) [inline src/pages/gen2/start/quickstart/nextjs-app-router/index.mdx:321]: the `as any` shouldn't be needed here. are you running into type errors?

File: src/pages/gen2/start/quickstart/nextjs-app-router/index.mdx [hunk 2]
```diff
@@ -0,0 +1,519 @@
...
+}
+```
+
+In `app/page.tsx`, update with the following code which import Amplify UI and wraps your App in the `withAuthenticator` function:
+
+```ts title="app/page.tsx"
+// app/page.tsx
+"use client";
+
+import { withAuthenticator } from "@aws-amplify/ui-react";
```
Comment (josefaidt) [inline src/pages/gen2/start/quickstart/nextjs-app-router/index.mdx:366]: nit, double quotes vs single quotes (used in the following example)

File: src/pages/gen2/start/quickstart/nextjs-app-router/index.mdx [hunk 3]
```diff
@@ -0,0 +1,519 @@
...
+
+Now, let's display data on our app's frontend. In the `components` directory create a `TodoList.tsx` file within it with the following code:
+
+```tsx title="components/TodoList.tsx"
+// components/TodoList.tsx
+'use client';
+
+import { useState, useEffect } from 'react';
+import { generateClient } from 'aws-amplify/data';
+import { Schema } from '@/amplify/data/resource';
```
Comment (josefaidt) [inline src/pages/gen2/start/quickstart/nextjs-app-router/index.mdx:392]:
```suggestion
import type { Schema } from '@/amplify/data/resource';
```

File: src/pages/gen2/start/quickstart/nextjs-pages-router/index.mdx [hunk 4]
```diff
@@ -0,0 +1,445 @@
...
+
+### View list of to-do items
+
+Now, let's display data on our app's frontend. Modify your app's home page file, `pages/index.tsx`, with the following code:
+
+```tsx title="pages/index.tsx"
+// pages/index.tsx
+import { useState, useEffect } from 'react';
+import { generateClient } from 'aws-amplify/data';
+import { Schema } from '@/amplify/data/resource';
```
Comment (josefaidt) [inline src/pages/gen2/start/quickstart/nextjs-pages-router/index.mdx:341]:
```suggestion
import type { Schema } from '@/amplify/data/resource';
```

File: src/pages/gen2/start/quickstart/nextjs-app-router/index.mdx [hunk 5]
```diff
@@ -0,0 +1,533 @@
...
+```
+
+Next, create a `components` folder in the root of your project and the contents below to a file called `ConfigureAmplify.tsx`.
+
+```ts title="components/ConfigureAmplify.tsx"
+// components/ConfigureAmplify.tsx
+
+"use client";
+
+import config from "@/amplifyconfiguration.json";
```
Comment (josefaidt) [inline src/pages/gen2/start/quickstart/nextjs-app-router/index.mdx:322]:
```suggestion
"use client";
import config from "@/amplifyconfiguration.json";
```

remove extra spacing

File: src/pages/gen2/start/quickstart/nextjs-app-router/index.mdx [hunk 6]
```diff
@@ -0,0 +1,533 @@
...
+
+In `app/page.tsx`, import `components/TodoList` and render it:
+
+```ts title="app/page.tsx"
+// app/page.tsx
+"use client";
+
+import { withAuthenticator } from "@aws-amplify/ui-react";
+import "@aws-amplify/ui-react/styles.css";
+
```
Comment (josefaidt) [inline src/pages/gen2/start/quickstart/nextjs-app-router/index.mdx:435]:
```suggestion
```

File: src/pages/gen2/start/quickstart/nextjs-app-router/index.mdx [hunk 7]
```diff
@@ -0,0 +1,533 @@
...
+  );
+}
+```
+
+Create a couple of to-dos, then refresh the page to see them. You can also subscribe to new to-dos in your `useEffect` to have them live reload on the page.
+
+```js
+useEffect(() => {
+  const sub = client.models.Todo.observeQuery().subscribe(({ items }) =>
+   setTodos([...items])
```
Comment (josefaidt) [inline src/pages/gen2/start/quickstart/nextjs-app-router/index.mdx:488]:
```suggestion
    setTodos([ ...items ])
```

small spacing adjustment

File: src/pages/gen2/start/quickstart/nextjs-pages-router/index.mdx [hunk 8]
```diff
@@ -0,0 +1,459 @@
...
+  );
+}
+```
+
+Create a couple of to-dos, then refresh the page to see them. You can also subscribe to new to-dos in your `useEffect` to have them live reload on the page.
+
+```js
+useEffect(() => {
+  const sub = client.models.Todo.observeQuery().subscribe(({ items }) =>
+   setTodos([...items])
```
Comment (josefaidt) [inline src/pages/gen2/start/quickstart/nextjs-pages-router/index.mdx:414]:
```suggestion
    setTodos([...items])
```

**Review by josefaidt — APPROVED** [review]

**Review by katiegoines — APPROVED** [review]

**Review by renebrandel — APPROVED** [review]

**Pull request merged**